src/components/forms/Checkbox.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import CheckEmptyIcon from '../../../lib/icons/react/CheckEmptyIcon';
+import CheckFilledIcon from '../../../lib/icons/react/CheckFilledIcon';
+import CheckIndeterminateIcon from '../../../lib/icons/react/CheckIndeterminateIcon';
+
+/**
+ * @param {Object} props react props
+ * @returns {ReactElement}
+ */
+const Checkbox = ({ name, label, value, onChange, indeterminate, checked, disabled }) => {
+  const [isChecked, setIsChecked] = useState(checked);
+
+  const handleChange = () => {
+    if (!disabled) {
+      const newState = !isChecked;
+      setIsChecked(newState);
+      onChange(newState);
+    }
+  };
+
+  let Icon = CheckEmptyIcon;
+  if (indeterminate && isChecked) {
+    Icon = CheckIndeterminateIcon;
+  } else if (isChecked) {
+    Icon = CheckFilledIcon;
+  }
+
+  return (
+    <div className={cx('fdsCheckbox', { 'fdsCheckbox--disabled': disabled })}>
+      <input
+        ref={(el) => {
+          /* eslint-disable no-unused-expressions,no-param-reassign */
+          el && isChecked && (el.indeterminate = indeterminate);
+          /* eslint-enable */
+        }}
+        type="checkbox"
+        name={name}
+        id={name}
+        value={value}
+        className="media--xs"
+        onChange={handleChange}
+        checked={isChecked}
+        disabled={disabled}
+      />
+      <label className="flush--bottom" htmlFor={name}>
+        <div role="checkbox" aria-checked={isChecked}>
+          <Icon size="xs" />
+        </div>
+        {label && <span className="padding--left--half">{label}</span>}
+      </label>
+    </div>
+  );
+};
+
+Checkbox.defaultProps = {
+  value: undefined,
+  checked: false,
+  indeterminate: false,
+  disabled: false,
```

Comment:
are the defaultProps here for explicitness, or specifically because undefined doesn't work in some expected way.

Reply:
I felt it was easier to reason about. It's really only there to be explicit.